### ISSUE TITLE
Comment newlines not working

### ISSUE THREAD

**Issue by Bammerbom** [SpongePowered/configurate#74]
Let's say I have a hocon configuration file which looks like this:
http://pastebin.com/CSSAbnse

If I use getComment() on "thing" it returns " Test Test2 Test3".
Configurate does not detect the new lines correctly, which it should.
New lines should be supported considering the javadocs say "If the comment contains multiple lines, the lines will be split by \n"

Related forum post: https://forums.spongepowered.org/t/problems-with-configurate/17137

**Comment by FerusGrim** [comment]
So, after a bit of testing to narrow down the point of the problem, I can for sure say that the problem occurs immediately after loading the configuration.

So it's the extraction from the source that's the issue - not a bug in how the information is saved.

That's useful.

Will update if I find anything else.

**Comment by FerusGrim** [comment]
So from what I can tell, the end of the line for comments within where I know the issue occurs is in `HoconConfigurationLoader#readConfigValue`. At which point, this line is fired:

```java
node.setComment(CRLF_MATCH.matcher(Joiner.on('\n').join(value.origin().comments())).replaceAll(""));
```

I've double-checked this line, and it doesn't exactly what it's supposed to. Takes a list of comments and joins them into a single line, with a `\n` between them. This is fine.

So it's somewhere _before_ this point where the issue occurs. So I've narrowed down where the issue occurs a bit further.

Will update again if I find more. 

I'm more or less keeping a log of what I find so that if I have to take a break, I know where to pick back up. Hope no one minds.

**Comment by FerusGrim** [comment]
I believe I've traced the issue to TypeSafe Config, which is way beyond what I'm capable of doing tonight. I may look into this tomorrow, but TypeSafe Config is quite a bit more complicated, I believe, than Configurate.

Relevant lines: `HoconConfigurationLoader#loadInternal`

```java
    @Override
    public void loadInternal(CommentedConfigurationNode node, BufferedReader reader) throws IOException {
        Config hoconConfig = ConfigFactory.parseReader(reader, parse);
        hoconConfig = hoconConfig.resolve();
        for (Map.Entry<String, ConfigValue> ent : hoconConfig.root().entrySet()) {
            readConfigValue(ent.getValue(), node.getNode(ent.getKey()));
        }
    }
```

From what I can tell, everything is working fine up until `loadInternal` is called. And everything after it checks out.

Meaning the `ConfigFactor.parseReader` call is the only thing that's suspicious.

The last hope for this to be a Configurate problem is if `ConfigParseOptions.defaults()` can be tweaked to allow multi-line comments.

**Comment by Meronat** [comment]
@lucko @FerusGrim Did https://github.com/SpongePowered/configurate/pull/83 solve this?

**Comment by lucko** [comment]
Yes, this can be closed

**Comment by Meronat** [comment]
Fixed by 731f7c19f88732b34de355d2a2306a57b75fd7ff
